src/components/Forms/LoginForm/LoginForm.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import style from './LoginForm.module.css';
+
+export class LoginForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      formErrors: { email: '', password: '' },
+      emailValid: false,
+      passwordValid: false,
+      formValid: false,
+    };
+  }
+
+  handleChange = ({ target }) => {
+    const { name, value } = target;
+    this.setState({ [name]: value });
+    this.validateField(name, value);
+  };
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { email, password } = this.state;
+    const { handleLogin } = this.props;
+    handleLogin(email, password);
+  };
+
+  handleFocus = ({ target }) => {
+    const { name, value } = target;
+    this.validateField(name, value);
+  };
+
+  validateField(fieldName, value) {
+    const { formErrors } = this.state;
+    let { emailValid, passwordValid } = this.state;
+    switch (fieldName) {
+      case 'email':
+        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
```

Comment:
all regExp should be kept in constants file

Reply:
```suggestion
        emailValid = value.match(REGEXP_KEYS.email);
```